tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,256 @@
+    "qpid::router::open_file_limit: 150100",
+    " ",
+    "postgresql::server::config_entries:",
+    "  max_connections: 1000",
+    "  shared_buffers: 4GB",
+    "  work_mem: 4MB",
+    "  checkpoint_segments: 32",
+    "  checkpoint_completion_target: 0.9",
+    "  effective_cache_size: 16GB",
+    "  autovacuum_vacuum_cost_limit: 2000",
+    "  log_min_duration_statement: 500 "
+]
+
+MMPV1_MONGODB = [
+    "# Added by foreman-installer during upgrade, run the "
+    "installer with --upgrade-mongo-storage to upgrade to WiredTiger.",
+    "mongodb::server::storage_engine: 'mmapv1'"
+]
+
+WIREDTIGER_MONGODB = [
+    "# Do not remove",
+    "mongodb::server::storage_engine: 'wiredTiger'"
+]
+
+
+@destructive
+class ScenarioPerformanceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer was able to change
+    the size or not
+
+    """
+    @staticmethod
+    def _custom_hiera_file_creation(mongodb_type, tune_size="default"):
+        """
+        This function we will use to create a custom-hiera.yaml file
+        as per users requirement"
+        :param int mongodb_type: Use to select  MMPV1 or WiredTiger data.
+        :param str tune_size: tune size would be medium, large, extra-large,
+        extra-extra-large.
+        """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("\n".join(DEFAULT_CUSTOM_HIERA_DATA))
+            if mongodb_type:
+                fd.write("\n".join(MMPV1_MONGODB))
+            else:
+                fd.write("\n".join(WIREDTIGER_MONGODB))
+            if tune_size == "medium":
+                fd.write("\n".join(MEDIUM_TUNING_DATA))
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command(cmd).return_code
+        self._custom_hiera_file_creation(mongodb_type, "medium")
+
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = ssh.command('satellite-installer --disable-system-checks',
+                                     connection_timeout=1000).stdout
+        command_output = [data.strip(" ") for data in command_output]
+        assert "Success!" in command_output
+
+    @post_upgrade
```

Comment:
@post_upgrade(depend_on=test_pre_performance_tuning_apply)

Reply:
Fixed.
And added an exception in pre-upgrade to keep the systems original state.